### PR TITLE
Blazor full stack web UI

### LINKED ISSUE #18289: Blazor Full-Stack Web App UI Option
We will work on this in this milestone to see if we can create an option on new solution creation.

### PR DESCRIPTION
**Pull request by maliming**
Resolve #18289

The document: 
https://github.com/abpframework/abp/blob/blazor-full-stack-web-ui/docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md

Test the CLI and new templates manually

- [x] app blazor wasm - leptonx-lite
- [x] app blazor server - leptonx-lite
- [x] app blazor server tiered - leptonx-lite
- [x] app blazor webapp - leptonx-lite
- [x] app blazor webapp tiered - leptonx-lite
- [x] app-nolayers blazor wasm - leptonx-lite
- [x] app-nolayers blazor server - leptonx-lite
- [x] module blazor - leptonx-lite
- [x] app blazor wasm - basic theme
- [x] app blazor server - basic theme
- [x] app blazor server tiered - basic theme
- [x] app blazor webapp - basic theme
- [x] app blazor webapp tiered - basic theme
- [x] app-nolayers blazor wasm - basic theme
- [x] app-nolayers blazor server - basic theme

### CODE REVIEW

**Comment by GadiZimerman** [comment]
@CodiumAI-Agent /describe

**Comment by GadiZimerman** [comment]
@CodiumAI-Agent /review

**Comment by jonathan-james-dev** [comment]
Do you know when this will be released? Is it possible to build the entire code base from your repo and get started that way? I really want to use the new template :)

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
+# New Blazor Web App Template in ABP 8.2
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
# Migrating to Blazor Web App
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+ASP.NET Blazor in .NET 8 allows you to use a single powerful component model to handle all of your web UI needs, including server-side rendering, client-side rendering, streaming rendering, progressive enhancement, and much more!
+
+ABP 8.2 supports the new Blazor Web App template, In this guide, we will introduce some new changes and features in the new Blazor Web App template.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
ABP v8.2.x supports the new Blazor Web App template, in this guide, we will introduce some new changes and features in the new Blazor Web App template.
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+## Create a new Blazor Web App
+
+> Please make sure you have installed the 8.2.x version of the ABP CLI.
+
+You can create a new Blazor Web App using the `abp new BookStore -t app -u blazor-webapp` command. The `-u blazor-webapp` option is used to select the Blazor Web App template.
+
+Of course, you can also create Blazor wasm and Blazor Server applications. We have changed them to use the new Blazor Web App mode.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
Of course, you can also create Blazor WASM and Blazor Server applications. We have changed them to use the new Blazor Web App mode:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+|      Type      |      Render mode
+|----------------|------------------
+|      WASM      |   InteractiveWebAssembly(prerender: false)
+|      Server    |   InteractiveServer
+|      WebApp    |   InteractiveAuto
+
+## The key changes of the new Blazor Web App template
+
+The new Web App template has two project, each of them contains a system of abp modules.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
The new Web App template has two projects, each containing a system of [ABP modules](https://docs.abp.io/en/abp/latest/Modules/Index).
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+## The key changes of the new Blazor Web App template
+
+The new Web App template has two project, each of them contains a system of abp modules. 
+
+- MyCompanyName.MyProjectName.Blazor.WebApp
+- MyCompanyName.MyProjectName.Blazor.WebApp.Client
+
+### MyCompanyName.MyProjectName.Blazor.WebApp
+
+The `Blazor.WebApp` is the startup project, There is a `App.razor` component in the `Blazor.WebApp` project, which is the root component of the Blazor application.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
The `Blazor.WebApp` is the startup project, and there is an `App.razor` component in the `Blazor.WebApp` project, which is the root component of the Blazor application.
```

**Comment by enisn** [comment]
We tried to create Blazor WebApp with local CLI build:

```
 dotnet volo.Abp.Cli.dll new BlazorWebAppTest -t app --version 8.1.0-rc.1 -csf -tv -u blazor-webapp
```

There are no Blazor Web App projects in the solution
image

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+The main differences between it and a regular blazor server project are:
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
The main differences between it, and a regular Blazor server project are:
```

**Comment by maliming** [comment]
Could you replace your local template zip with the current branch?

`C:\Users\username\.abp\templates`

`abp new BookStore -t app -u blazor-webapp` 
The `-u blazor-webapp` option is used to select the Blazor Web App template.

see https://github.com/abpframework/abp/blob/blazor-full-stack-web-ui/docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md

One thing to note: **abp adds a new class, while lepton needs to use it, but lepton is using package references.**

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+
+1. You need to `PreConfigure` the `AbpAspNetCoreComponentsWebOptions's IsBlazorWebApp` to `true`.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
1. You need to `PreConfigure` the `IsBlazorWebApp` to `true` in `AbpAspNetCoreComponentsWebOptions`:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+public override void PreConfigureServices(ServiceConfigurationContext context)
+{
+    PreConfigure<AbpAspNetCoreComponentsWebOptions>(options =>
+    {
+        options.IsBlazorWebApp = true;
+    });
+}
+````
+
+2. Add related services to the container. Add `MyProjectNameBlazorClientModule's` assembly to the `AbpRouterOptions's AdditionalAssemblies`.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
2. Add related services to the container. Add assembly of `MyProjectNameBlazorClientModule` to the `AdditionalAssemblies` by configuring `AbpRouterOptions`:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md (hunk 1 above)
Comment (berkansasmaz) [reply docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]: Also, change the file name to "Migrating-to-Blazor-Web-App"

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+            .AddAdditionalAssemblies(builder.ServiceProvider.GetRequiredService<IOptions<AbpRouterOptions>>().Value.AdditionalAssemblies.ToArray());
+    });
+}
+````
+
+### MyCompanyName.MyProjectName.Blazor.WebApp.Client
+
+There is a `Routers.razor` component in the `Blazor.WebApp.Client` project, which is used by the `App.razor` component.
+
+The main differences between it and a regular blazor wasm project are:
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
The main differences between it and a regular Blazor WASM project are:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+
+1. You need to `PreConfigure` the `AbpAspNetCoreComponentsWebOptions's IsBlazorWebApp` to `true`.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
1. You need to `PreConfigure` the `IsBlazorWebApp` to `true` in `AbpAspNetCoreComponentsWebOptions`:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+public override void PreConfigureServices(ServiceConfigurationContext context)
+{
+    PreConfigure<AbpAspNetCoreComponentsWebOptions>(options =>
+    {
+        options.IsBlazorWebApp = true;
+    });
+}
+````
+
+2. Use `AddBlazorWebAppServices` to replace `Authentication` code.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
2. Use `AddBlazorWebAppServices` to replace `Authentication` code:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+
+3. Remove the `builder.RootComponents.Add<App>("#ApplicationContainer");` code.
+
+### MyCompanyName.MyProjectName.Blazor.WebApp.Tiered and MyCompanyName.MyProjectName.Blazor.WebApp.Tiered.Client
+
+The tiered projects are the same as the WebApp projects, but the authentication configuration is different.
+
+We need share the `access_token` to `Client` project.
+
+`App.rzor` of `MyCompanyName.MyProjectName.Blazor.WebApp.Tiered`:
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
Add code block to `App.razor` of `MyCompanyName.MyProjectName.Blazor.WebApp.Tiered` as below:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 13]
```diff
@@ -0,0 +1,223 @@
...
+                AccessToken = Token
+            });
+        }
+
+        await Task.CompletedTask;
+    }
+}
+````
+
+`MyProjectNameBlazorClientModule` of `MyCompanyName.MyProjectName.Blazor.WebApp.Tiered.Client`:
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
Add `ConfigureAuthentication` to `MyProjectNameBlazorClientModule` of `MyCompanyName.MyProjectName.Blazor.WebApp.Tiered.Client` as below:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 14]
```diff
@@ -0,0 +1,223 @@
...
+      "InteractiveAuto": true,
+      "Parameters": {
+
+      }
+    }
+  }
+}
+````
+
+For Blazor wasm and Blazor Server applications, you need to set `IsBlazorWebApp` to `true` and not need change the `InteractiveAuto`.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
For Blazor WASM and Blazor Server applications, you need to set `IsBlazorWebApp` to `true` and not need to change the `InteractiveAuto`:
```

File: docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md [hunk 15]
```diff
@@ -0,0 +1,223 @@
...
+````csharp
+private static void ConfigureAuthentication(WebAssemblyHostBuilder builder)
+{
+    builder.Services.AddBlazorWebAppTieredServices();
+}
+````
+
+## ABP Bundle
+
+You need set `IsBlazorWebApp` and `InteractiveAuto` to `true` in the `appsettings.json` file of the `MyCompanyName.MyProjectName.Blazor.WebApp.Client` project.
```
Comment (berkansasmaz) [inline docs/en/Migration-Guides/Abp-8-2-Blazor-Web-App.md]:
```suggestion
You need set `IsBlazorWebApp` and `InteractiveAuto` to `true` in the `appsettings.json` file of the `MyCompanyName.MyProjectName.Blazor.WebApp.Client` project:
```

**Review by berkansasmaz — CHANGES_REQUESTED** [review]
Add a semicolon before every code block as below:

[Your sentence]: 

```csharp
code block
```

**Pull request merged**